dockstore-client/src/test/java/io/dockstore/client/cli/YamlValidatorTest.java

```diff
@@ -248,6 +231,55 @@ void dotGitHubDirectory() {
             assertEquals(errorMsg, ex.getMessage());
         }
     }
+
+    @Test
+    void invalidSubclass() {
+        String testDirectory = "src/test/resources/YamlVerifyTestDirectory/invalid-subclass";
+        try {
+            YamlVerifyUtility.dockstoreValidate(testDirectory);
+        } catch (YamlVerifyUtility.ValidateYamlException ex) {
+            String errorMsg = YamlVerifyUtility.INVALID_DOCKSTORE_YML + testDirectory + "/" + YamlVerifyUtility.DOCKSTOREYML
+                    + YamlVerifyUtility.CONTAINS_ERRORS
+                    // The below part of the error message is generated by this line,
+                    // https://github.com/dockstore/dockstore-cli/blob/5aac6e6d221b7618bc9769a95834f73f37f75aaf/dockstore-client/src/main/java/io/dockstore/client/cli/YamlVerifyUtility.java#L188
+                    + "Property \"workflows[0].subclass\" must be a supported descriptor language (\"CWL\", \"WDL\", "
+                    + "\"GALAXY\", or \"NFL\") (current value: \"INVALID_SUBCLASS\")";
+            assertEquals(errorMsg, ex.getMessage());
+        }
+    }
+
+
+    @Test
+    void noPrimaryDescriptorPath() {
+        final String baseTestDirectory = "src/test/resources/YamlVerifyTestDirectory/no-primary-descriptor-path/";
+        List<String> directoryEnds = Arrays.asList(TOOL, WORKFLOW);
+        for (String directoryEnd : directoryEnds) {
+            try {
+                String testDirectory = baseTestDirectory + directoryEnd;
+                YamlVerifyUtility.dockstoreValidate(testDirectory);
+                fail("A .dockstore.yml that does not contain as primary descriptor path passed");
+            } catch (YamlVerifyUtility.ValidateYamlException ex) {
+                assertTrue(ex.getMessage().contains("primaryDescriptorPath\" is missing but required"));
+            }
+        }
+    }
+
+    @Test
+    void emptyPrimaryDescriptorPath() {
+        final String baseTestDirectory = "src/test/resources/YamlVerifyTestDirectory/empty-primary-descriptor-path/";
+        List<String> directoryEnds = Arrays.asList(TOOL, WORKFLOW);
```

Comment:
This code block looks identical to the code starting on line 255; you could create a method out of it.